### PR TITLE
docs: Updated design document page

### PR DESCRIPTION
**Pull request by iamrajiv**
- [ ] I added CHANGELOG entry for this change.
- [X] Change is not relevant to the end user.

## Changes

- Fixed grammatical mistakes.
- Formated markdown file.

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -63,29 +62,32 @@ The meta.json is updated during upload time on sidecars.
...
 
 ### Stores
 
 A store node acts as a gateway to block data that is stored in an object storage bucket. It implements the same gRPC API as data sources to provide access to all metric data found in the bucket.
 
 It continuously synchronizes which blocks exist in the bucket and translates requests for metric data into object storage requests. It implements various strategies to minimize the number of requests to the object storage such as filtering relevant blocks by their metadata (e.g. time range and labels) and caching frequent index lookups.
 
-The Prometheus 2.0 storage layout is optimized for minimal read amplification. For example, sample data for the same time series is sequentially aligned in a [chunk file](design.md/#chunk-file). Similarly, series for the same metric name are sequentially aligned as well.
+The Prometheus 2.0 storage layout is optimized for minimal read amplification. For example, sample data for the same time series are sequentially aligned in a [chunk file](design.md/#chunk-file). Similarly, series for the same metric name is sequentially aligned as well.
```
Comment (bwplotka) [inline docs/design.md:80]: Sample data `is` not `are` I think? (: The same in second sentence `series` is used here as single series.

File: docs/design.md [hunk 2]
```diff
@@ -137,33 +137,34 @@ Based on the metadata of store and source nodes, they attempt to minimize the re
...
+Store, rule, and compactor nodes are all expected to scale significantly within a single instance or high availability pair. Similar to Prometheus, functional sharding can be applied for rare cases in which this does not hold.
 
 For example, rule sets can be divided across multiple HA pairs of rule nodes. Store nodes likely are subject to functional sharding regardless by assigning dedicated buckets per region/datacenter.
 
 Overall, first-class horizontal sharding is possible but will not be considered for the time being since there's no evidence that it is required in practical setups.
 
 ## Cost
 
-The only extra cost Thanos adds to an existing Prometheus setup is essentially the price of storing and querying data from the object storage and running of the store node.
+The only extra cost Thanos adds to an existing Prometheus setup is essentially the price of storing and querying data from the object storage and running of the storage node.
```
Comment (bwplotka) [inline docs/design.md:156]:
```suggestion
The only extra cost Thanos adds to an existing Prometheus setup is essentially the price of storing and querying data from the object storage and running of the `store` node.
```

We actually mean Thanos Store GW

**Review by bwplotka — APPROVED** [review]
Thank you! Some comments otherwise LGTM (:

**Comment by Namanl2001** [comment]
@iamrajiv You need to sign off each of your commits to pass the DCO check. Thanks

**Review by kakkoyun — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**